### PR TITLE
[UI] Getting started page for hosted pipelines

### PR DESCRIPTION
**Pull request by Bobgy**
UPDATE: also implemented layout similar to designed documentation
UPDATE2: adjusted page title
localhost_3000_ (1)
Content of this page is purely rendered from markdown here: https://github.com/kubeflow/pipelines/pull/2935/files#diff-1e8357336f0f74b158fbf0751ac2d9daR31. So it's super easy for anyone to update later.

demo when feature flag turned on: https://drive.google.com/file/d/12dyaTLQY3AvwG38YC3TmBhCGOYqSUmGg/view

Changes:
1. Add a separate getting started page for tfx usage (~~exact content of this page is merely a copy from https://github.com/kubeflow/pipelines/pull/2928~~ implemented requested layout, please adjust content with markdown based on further discussion).
2. Add a feature flag for pipelines in gcp marketplace. (ml-pipeline-ui deployment with env variable DEPLOYMENT=MARKETPLACE will turn on this new feature.)
3. Only show 1. when the feature flag is on.

Feature flag verification:
1. built a frontend image using this branch
2. swapped the image in a ml-pipeline-ui deployment and changed env var to DEPLOYMENT=MARKETPLACE
3. verified getting started page shows
4. edited deployment again to change env var to DEPLOYMENT=KUBEFLOW
5. verified getting started page is hidden

This supercedes https://github.com/kubeflow/pipelines/pull/2928.

/assign @SinaChavoshi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2935)

### CODE REVIEW

**Comment by Bobgy** [comment]
/test kubeflow-pipeline-sample-test

strange, flakiness rate of workload identity seems to be higher than before

File: frontend/src/pages/GettingStarted.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    };
+  }
+
+  public async refresh() {
+    // do nothing
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div className={classes(commonCss.page, padding(20, 'lr'))}>
```
Comment (SinaChavoshi) [inline frontend/src/pages/GettingStarted.tsx]: Thank you so much for all the work ! this is very nicely done and an amazing turn around time!
Do you think there is any chance that you could match the lay out in option 0 ("in Templates, Demos ,and Tutorials UI design" for the content of this page.  if not no worries you already have done most of the heavy lifting,  I'm just worried about the short time remaining tomorrow.

Comment (Bobgy) [reply frontend/src/pages/GettingStarted.tsx]: Great, I'm glad it's helpful.
Also updated the start page to use markdown with some styling I just defined. I believe that will be the most easy to change for the team later.

Comment (SinaChavoshi) [reply frontend/src/pages/GettingStarted.tsx]: This is perfect ! thank you .

**Comment by sarahmaddox** [comment]
/assign @joeliedtke 

Thanks for asking for a tech writer review, @Ark-kun! I've assigned @joeliedtke, as he's now the tech writer for Pipelines.  (I'm continuing as the Kubeflow tech writer overall.)

**Comment by Ark-kun** [comment]
Thank you for adding this page!

The page looks nice and useful, but I've noticed that it misses a link to the Pipelines documentation (https://www.kubeflow.org/docs/pipelines/), the Quickstart, the Getting Started guide etc. We might want to link to that helpful information.

/cc @sarahmaddox @amygdala @paveldournov

**Comment by jessiezcc** [comment]
Thanks @Bobgy for this PR, it looks great! 
For pipelines page, does it only show user uploaded pipeline or will it still show demo/tutorial pipelines?

**Comment by Bobgy** [comment]
@Ark-kun, FYI, the page is currently only a mockup to demonstrate markdown. I expect a separate PR to adjust the exact content.

@jessiezcc, pipelines page still shows preloaded pipelines. I don't see how we can distinguish preloaded from user uploaded. So that's something I couldn't implement now.

**Comment by jessiezcc** [comment]
for now, can we differentiate using prefix such as [demo], [tutorial]? WDUT, @SinaChavoshi ?
this way, end user will be able to customer their demo/tutorial list by just using the naming convention as well.

**Comment by Bobgy** [comment]
@jessiezcc thanks for the idea. I think that's doable, but would need quite some efforts. I'm not sure I can fit that in current time frame.

Also, when I was trying to test filtering, found a bug with it: https://github.com/kubeflow/pipelines/issues/2945. We should fix this before beta.

**Comment by Bobgy** [comment]
Note, content of this getting started page is purely rendered from markdown here: https://github.com/kubeflow/pipelines/pull/2935/files#diff-1e8357336f0f74b158fbf0751ac2d9daR31. So it's super easy for anyone to update later.

I suggest we get this merged and let someone in US time work on the final tech writings. So that communication loop would be minimized. @SinaChavoshi can I ask you for that?

**Comment by jessiezcc** [comment]
agree, let's check in this first

**Comment by Bobgy** [comment]
@rmgogogo @IronPan for approval, because this includes marketplace manifest change too.

**Comment by SinaChavoshi** [comment]
Yuan thank you again for all the work, +1 lets get this in as is. We can do quick follow up to adjust the UI before build cut off date today. As Yuan mentioned changes from here will be trivial.

**Pull request merged**